### PR TITLE
📚 DOCS: Add developer tips and tricks section

### PR DESCRIPTION
**Pull request by mmcky**
This PR adds a small section with some helpful tips for getting access to a documents xml structure to visualise the sphinx.AST

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Developer Guide
```
Comment (chrisjsewell) [inline docs/developer.md:1]:
```suggestion
# Developement Tips and Tricks
```

File: docs/developer.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+This section provides some useful tips and tricks for development of projects
+within the Executable Book Project ecosystem.
+
+## Sphinx Tips and Tricks
```
Comment (chrisjsewell) [inline docs/developer.md:6]:
```suggestion
## Working with Sphinx
```

File: docs/developer.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Developer Guide
+
+This section provides some useful tips and tricks for development of projects
+within the Executable Book Project ecosystem.
```
Comment (chrisjsewell) [inline docs/developer.md:4]:
```suggestion
This section provides some useful tips and tricks for development of projects within the Executable Book Project ecosystem.
```

File: docs/developer.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Sphinx Tips and Tricks
+
+The [sphinx development guide](https://www.sphinx-doc.org/en/master/develop.html) is also a useful
+resource for understanding how `Sphinx` works.
```
Comment (chrisjsewell) [inline docs/developer.md:9]:
```suggestion
The [sphinx development guide](https://www.sphinx-doc.org/en/master/develop.html) is also a useful resource for understanding how `Sphinx` works.
```

File: docs/developer.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+### Getting Access to the Sphinx Abstract Syntax Tree (AST)
+
+Getting access to the `xml` representation of the abstract syntax tree (AST) is a very
+important step in understanding how Sphinx has organised the document.
+
+One way to get this information is from the `.doctree` directory
+contained in a project `_build` directory.
```
Comment (chrisjsewell) [inline docs/developer.md:17]:
```suggestion
One way to get this information is from the `.doctree` directory contained in a project `_build` directory.
```

File: docs/developer.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+One you have built a sample project you can get access to the AST by loading the pickled
+doctree in `python`:
```
Comment (chrisjsewell) [inline docs/developer.md:20]:
```suggestion
Once you have built a sample project you can get access to the AST by loading the pickled
doctree in `python`:
```

**Comment by mmcky** [comment]
@chrisjsewell thanks for your suggested changes. I have accepted all of them. Also when you commit `suggested` changes do you do that through the `web` interface (https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request) or can you do that via `vscode`/`commit` workflow?

**Comment by chrisjsewell** [comment]
Yeh I do it through web, and just add all the suggestion to a batch (as it shows in that link) then you can just do one commit rather than 6 😉
Haven't looked if thats available in vscode yet?

**Comment by choldgraf** [comment]
It'll be available soon enough whenever codespaces gets activated

**Review by chrisjsewell — APPROVED** [review]

**Pull request merged**